### PR TITLE
feat: add more validation for CronFederatedHPA

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
CronFedertedHPA validation enhancement:
1. `TargetReplicas` should be larger than or equal to 0.
2. `TargetMinReplicas` and `TargetMaxReplicas` should be larger than or equal to 1.
3. `TargetMinReplicas` should be larger than or equal to `TargetMaxReplicas`.
4. metadata.name should be a valid name.
5. spec.scaleTargetRef should be valid.

**Which issue(s) this PR fixes**:
Fixes #none

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-webhook`: Validation enhancement, including:
1. `TargetReplicas` should be larger than or equal to 0.
2. `TargetMinReplicas` and `TargetMaxReplicas` should be larger than or equal to 1.
3. `TargetMinReplicas` should be larger than or equal to `TargetMaxReplicas`.
4. metadata.name should be a valid name.
5. spec.scaleTargetRef should be valid.
```

### CODE REVIEW

File: pkg/webhook/cronfederatedhpa/validation.go [hunk 1]
```diff
@@ -64,69 +70,95 @@ func (v *ValidatingAdmission) InjectDecoder(d *admission.Decoder) error {
...
+	scaleTargetKind string, fldPath *field.Path) field.ErrorList {
+	errs := field.ErrorList{}
+
+	if scaleFHPA {
+		// Validate targetMinReplicas and targetMaxReplicas
+		if rule.TargetMinReplicas == nil && rule.TargetMaxReplicas == nil {
+			errs = append(errs, field.Invalid(fldPath.Child("targetMaxReplicas"), "",
+				"targetMinReplicas and targetMaxReplicas cannot be nil at the same time if you want to scale FederatedHPA"))
+		}
+		if rule.TargetMinReplicas != nil && *rule.TargetMinReplicas <= 0 {
```
Comment (whitewindmills) [inline pkg/webhook/cronfederatedhpa/validation.go:138]: try `if pointer.Int32Deref(rule.TargetMinReplicas, 1) <= 0`.

File: pkg/webhook/cronfederatedhpa/validation.go [hunk 2]
```diff
@@ -64,69 +70,95 @@ func (v *ValidatingAdmission) InjectDecoder(d *admission.Decoder) error {
...
+			errs = append(errs, field.Invalid(fldPath.Child("targetMinReplicas"), "",
+				"targetMinReplicas should be larger than 0"))
+		}
+		if rule.TargetMaxReplicas != nil && *rule.TargetMaxReplicas <= 0 {
+			errs = append(errs, field.Invalid(fldPath.Child("targetMaxReplicas"), "",
+				"targetMaxReplicas should be larger than 0"))
 		}
+		if rule.TargetMinReplicas != nil && rule.TargetMaxReplicas != nil && *rule.TargetMinReplicas > *rule.TargetMaxReplicas {
```
Comment (whitewindmills) [inline pkg/webhook/cronfederatedhpa/validation.go:146]: ditto, try to simplify it.

**Review by whitewindmills — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**